packages/grid/src/MockDataBarGridModel.ts

```diff
@@ -0,0 +1,202 @@
+type MarkerMap = Map<ModelIndex, Marker[]>;
+/** Map from column to whether the bar has a gradient */
+type GradientMap = Map<ModelIndex, boolean>;
+
+function isArrayOfNumbers(value: unknown): value is number[] {
+  return Array.isArray(value) && value.every(item => typeof item === 'number');
+}
+
+export function isDataBarGridModel(
+  model: GridModel
+): model is MockDataBarGridModel {
+  return (model as MockDataBarGridModel)?.columnAxes !== undefined;
+}
+
+class MockDataBarGridModel extends GridModel {
+  private numberOfColumns;
+
+  private numberOfRows;
+
+  private data: unknown[][];
+
+  private columnMin: MinMap;
+
+  private columnMax: MaxMap;
+
+  columnAxes: ColumnAxisMap;
+
+  valuePlacements: ValuePlacementMap;
+
+  directions: DirectionMap;
+
+  positiveColors: ColorMap;
+
+  negativeColors: ColorMap;
+
+  // Opacities should be between 0 and 1
+  opacities: OpacityMap;
+
+  textAlignments: TextAlignmentMap;
+
+  markers: MarkerMap;
+
+  hasGradients: GradientMap;
+
+  constructor(
+    data: unknown[][],
+    columnAxes = new Map(),
+    positiveColors = new Map(),
+    negativeColors = new Map(),
+    valuePlacements = new Map(),
+    opacities = new Map(),
+    directions = new Map(),
+    textAlignments = new Map(),
+    markers: MarkerMap = new Map(),
+    hasGradients: GradientMap = new Map()
+  ) {
+    super();
+
+    this.positiveColors = positiveColors;
+    this.negativeColors = negativeColors;
+    this.data = data;
+    this.columnAxes = columnAxes;
+    this.valuePlacements = valuePlacements;
+    this.opacities = opacities;
+    this.directions = directions;
+    this.textAlignments = textAlignments;
+    this.markers = markers;
+    this.numberOfRows = Math.max(...data.map(row => row.length));
+    this.numberOfColumns = data.length;
+    this.columnMin = new Map();
+    this.columnMax = new Map();
+    this.hasGradients = hasGradients;
+
+    for (let i = 0; i < data.length; i += 1) {
+      const column = data[i];
+      if (isArrayOfNumbers(column)) {
+        this.columnMin.set(i, Math.min(...column));
+        this.columnMax.set(i, Math.max(...column));
+      }
+    }
+  }
+
+  minOfColumn(column: ModelIndex): number {
+    return getOrThrow(this.columnMin, column);
+  }
+
+  maxOfColumn(column: ModelIndex): number {
+    return getOrThrow(this.columnMax, column);
+  }
+
+  get rowCount() {
+    return this.numberOfRows;
+  }
+
+  get columnCount() {
+    return this.numberOfColumns;
+  }
+
+  textForCell(column: number, row: number): string {
+    return `${this.data[column]?.[row]}`;
+  }
+
+  textForColumnHeader(column: number): string {
+    return `${column}`;
+  }
+
+  textAlignForCell(column: number, row: number): CanvasTextAlign {
+    return this.textAlignments.get(column) ?? DEFAULT_TEXT_ALIGNMENT;
+  }
+
+  rendererForCell(column: ModelIndex, row: ModelIndex): CellRendererType {
+    return 'databar';
+  }
+
+  barColorForCell(column: ModelIndex, row: ModelIndex): string {
```

Comment:
With Don's comment, this should be:
```suggestion
  barColorForCell(column: ModelIndex, row: ModelIndex): GridColor | [GridColor, GridColor] {
```
And then add TSDocs in the interface definition about how specifying multiple colours gives it a gradient.

Also note that I still think this should be combined with all the options for the databar (e.g. `this.dataBarForCell`)